tests/compute/test_sparse.py

```diff
@@ -0,0 +1,69 @@
+import dgl
+import networkx as nx
+import backend as F
+
+udf_msg = {
+    'add': lambda edges: {'m': edges.src['x'] + edges.data['w']},
+    'sub': lambda edges: {'m': edges.src['x'] - edges.data['w']},
+    'mul': lambda edges: {'m': edges.src['x'] * edges.data['w']},
+    'div': lambda edges: {'m': edges.src['x'] / edges.data['w']},
+    'copy_u': lambda edges: {'m': edges.src['x']},
+    'copy_e': lambda edges: {'m': edges.data['w']}
+}
+
+udf_apply_edges = {
+    'add': lambda edges: {'m': edges.src['x'] + edges.dst['y']},
+    'sub': lambda edges: {'m': edges.src['x'] - edges.dst['y']},
+    'mul': lambda edges: {'m': edges.src['x'] * edges.dst['y']},
+    'div': lambda edges: {'m': edges.src['x'] / edges.dst['y']},
+    'dot': lambda edges: {'m': F.sum(edges.src['x'] * edges.dst['y'], -1, keepdims=True)},
+    'copy_u': lambda edges: {'m': edges.src['x']},
+}
+
+udf_reduce = {
+    'sum': lambda nodes: {'v': F.sum(nodes.mailbox['m'], 1)},
+    'min': lambda nodes: {'v': F.min(nodes.mailbox['m'], 1)},
+    'max': lambda nodes: {'v': F.max(nodes.mailbox['m'], 1)}
+}
+
+def test_spmm():
```

Comment:
DId you add test for broadcast?

Reply:
We are testing on a broadcasting setting.